### PR TITLE
feat(suite): app promo banners

### LINKED ISSUE #8234: Promote mobile app and desktop suite CTA
## Design

* [Dashboard](https://www.figma.com/file/YoGkN5kl5FUdxZ2EOkpZTL/Exploring-Navigation?node-id=1440%3A8088&t=7NsJHbUZ1Joee0KW-1)
* [Settings](https://www.figma.com/file/YoGkN5kl5FUdxZ2EOkpZTL/Exploring-Navigation?node-id=1201-4670&t=ewPgTXI14dQoIXFO-4)

## Spec

Historical product specs
* https://www.notion.so/satoshilabs/Trezor-Suite-Lite-CTA-2588f3c57db947f9816e3e26744f47db
* https://www.notion.so/satoshilabs/Download-desktop-suite-CTAs-for-web-users-0609d608426e4a9e8e1e7fac44630469

As a web suite user
I want to download Trezor suite
So that I am secure

Given I am on the dashboard
When I scroll to the bottom
Then I should see a 'Download desktop app' button
And a Google Play store button
And an App Store store button

As a desktop suite user
I want to download Trezor Suite Lite
So that I can monitor my balance without a computer

Given I am on the dashboard
When I scroll to the bottom
Then I should see a Google Play store button
And an App Store store button

## Preflight checklist
- [x] Product/Design
- [x] [App store links](https://www.notion.so/satoshilabs/Mobile-app-general-info-6d4c382f42f74d8bb9bea1124c28ffff)
- [x] Copy
- [x] Tech Feedback

### PR DESCRIPTION
**Pull request by dahaca**
## Description

* add a promo footer on the dashboard page
* add a promo banner to the settings page

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/8234

## Screenshots:

image

image

### CODE REVIEW

File: packages/urls/src/urls.ts [hunk 1]
```diff
@@ -23,6 +23,10 @@ export const SUITE_BRIDGE_URL = 'https://suite.trezor.io/web/bridge/';
 export const SUITE_FIRMWARE_URL = 'https://suite.trezor.io/web/firmware/';
 export const SUITE_UDEV_URL = 'https://suite.trezor.io/web/udev/';
 
+export const SUITE_MOBILE_APP_STORE = 'https://apps.apple.com/cz/app/sonic-racing/id1461172325';
+export const SUITE_MOBILE_PLAY_STORE =
+    'https://play.google.com/store/apps/details?id=com.sega.sonic1px&hl=en&gl=US';
```
Comment (dahaca) [inline packages/urls/src/urls.ts]: ### FAKE URLS FOR TESTING PURPOSES!!!

Comment (dahaca) [reply packages/urls/src/urls.ts]: e888f4320

File: packages/suite/src/components/suite/SuiteLayout/index.tsx [hunk 2]
```diff
@@ -68,19 +72,20 @@ const AppWrapper = styled.div`
...
+    padding: 24px ${DESKTOP_HORIZONTAL_PADDINGS} 90px ${DESKTOP_HORIZONTAL_PADDINGS};
 
     ${variables.SCREEN_QUERY.BELOW_LAPTOP} {
-        padding: 24px 16px 70px 16px;
+        padding: 24px ${MOBILE_HORIZONTAL_PADDINGS} 70px ${MOBILE_HORIZONTAL_PADDINGS};
     }
 
     ${variables.SCREEN_QUERY.MOBILE} {
-        padding-bottom: 50px;
+        padding-bottom: 90px; // including the mobile app banner
```
Comment (komret) [inline packages/suite/src/components/suite/SuiteLayout/index.tsx]: Linter does not like this type of comments with `styled-components`, use `/* */` instead.

File: packages/suite/src/views/dashboard/components/PromoBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    padding: 4px;
+    background-color: white;
+    border-radius: 6px;
+`;
+
+type QrType = 'app-store' | 'play-store';
+
+interface StoreBadgeProps {
+    url: string;
+    image: string;
```
Comment (komret) [inline packages/suite/src/views/dashboard/components/PromoBanner.tsx]: Removes the need for type casting:

```suggestion
    image: 'APP_STORE' | 'PLAY_STORE';
```

File: packages/suite/src/views/dashboard/components/PromoBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    type: QrType;
+    shownQRState: [QrType | undefined, (type: QrType | undefined) => void];
+}
+
+const StoreBadge = ({ url, image, type, shownQRState: [showQR, setShowQr] }: StoreBadgeProps) => {
```
Comment (komret) [inline packages/suite/src/views/dashboard/components/PromoBanner.tsx:133]: Could be a separate file, but either way is OK.

File: packages/suite/src/views/dashboard/components/PromoBanner.tsx [hunk 5]
```diff
@@ -0,0 +1,209 @@
+import React, { useState } from 'react';
+import styled, { css } from 'styled-components';
+import { SUITE_MOBILE_APP_STORE, SUITE_MOBILE_PLAY_STORE, SUITE_URL } from '@trezor/urls';
```
Comment (sime) [inline packages/suite/src/views/dashboard/components/PromoBanner.tsx:3]: Nitpick optimisation, can we dive into the library and update SUITE_URL to `https://trezor.io/trezor-suite`. As the current URL redirects there.

File: packages/suite/src/views/dashboard/components/PromoBanner.tsx [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    :focus {
+        background: ${({ theme }) => theme.STROKE_GREY};
+        opacity: 1;
+    }
+`;
+
+const BadgeContainer = styled.div`
+    display: flex;
+    gap: 10px;
+    margin-left: auto;
```
Comment (komret) [inline packages/suite/src/views/dashboard/components/PromoBanner.tsx]: I'd find this sort of alignment a bit clearer as it ties the message to the button:

Screenshot 2023-05-11 at 14 38 45

But again, either way is fine.

**Review by komret — CHANGES_REQUESTED** [review]

**Review by sime — CHANGES_REQUESTED** [review]
App store buttons on desktop react to hover with a QR code ✅ 

App store buttons do not react to click/touch event on desktop/mobile 🛑

File: packages/suite/src/views/dashboard/components/PromoBanner.tsx (hunk 6 above)
Comment (dahaca) [reply packages/suite/src/views/dashboard/components/PromoBanner.tsx]: was also considering that, @sime u ok with that?

Comment (sime) [reply packages/suite/src/views/dashboard/components/PromoBanner.tsx]: Works for me. How are we looking on browser view though?

Screenshot 2023-05-11 at 15 56 29

Comment (komret) [reply packages/suite/src/views/dashboard/components/PromoBanner.tsx]: Browser looks good to me. Just a bit worried the right message could expand to a third line in other languages if it is longer.

File: packages/suite/src/views/dashboard/components/PromoBanner.tsx (hunk 3 above)
Comment (dahaca) [reply packages/suite/src/views/dashboard/components/PromoBanner.tsx]: still needed, since i am using template literals :(

Comment (dahaca) [reply packages/suite/src/views/dashboard/components/PromoBanner.tsx]: [6649107](https://github.com/trezor/trezor-suite/pull/8435/commits/6649107ca83ba656674d9c5f155b25dfd87e1e0d)

File: packages/suite/src/components/suite/SuiteLayout/index.tsx (hunk 2 above)
Comment (dahaca) [reply packages/suite/src/components/suite/SuiteLayout/index.tsx]: [6649107](https://github.com/trezor/trezor-suite/pull/8435/commits/6649107ca83ba656674d9c5f155b25dfd87e1e0d)

File: packages/suite/src/views/dashboard/components/PromoBanner.tsx (hunk 5 above)
Comment (dahaca) [reply packages/suite/src/views/dashboard/components/PromoBanner.tsx:3]: [6649107](https://github.com/trezor/trezor-suite/pull/8435/commits/6649107ca83ba656674d9c5f155b25dfd87e1e0d)

File: packages/suite/src/support/messages.ts [hunk 7]
```diff
@@ -7950,4 +7950,24 @@ export default defineMessages({
         id: 'TR_INCLUDING_FEE',
         defaultMessage: 'Including fee',
     },
+    TR_DESKTOP_APP_PROMO_HEADING: {
+        id: 'TR_DESKTOP_APP_PROMO_HEADING',
+        defaultMessage: 'Get the most out of Trezor Suite',
+    },
+    TR_DESKTOP_APP_PROMO_TEXT: {
+        id: 'TR_INCLUDING_FEE',
```
Comment (komret) [inline packages/suite/src/support/messages.ts]: Wrong ID

Comment (dahaca) [reply packages/suite/src/support/messages.ts]: 58df7ee3c

File: packages/suite/src/views/dashboard/components/PromoBanner.tsx (hunk 6 above)
Comment (dahaca) [reply packages/suite/src/views/dashboard/components/PromoBanner.tsx]: 58df7ee3c

Comment (dahaca) [reply packages/suite/src/views/dashboard/components/PromoBanner.tsx]: Ideally the copywriters should try to fit the text but I could fix it later if it looks ugly :)

File: packages/suite/src/views/dashboard/components/PromoBanner.tsx (hunk 4 above)
Comment (dahaca) [reply packages/suite/src/views/dashboard/components/PromoBanner.tsx:133]: I believe the badge is small enough of a component and it's simpler to keep them together in this case 🤷‍♂️

File: packages/suite/src/views/dashboard/components/PromoBanner.tsx [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+    ${promoContainerCss}
+    border-right: 1px solid ${({ theme }) => theme.STROKE_GREY};
+`;
+
+const MobilePromoContainer = styled.div`
+    ${promoContainerCss}
+
+    > :first-child {
+        margin-left: auto;
+    }
```
Comment (komret) [inline packages/suite/src/views/dashboard/components/PromoBanner.tsx]:
```suggestion
    justify-content: flex-end;
```

**Review by komret — APPROVED** [review]

File: packages/suite/src/views/dashboard/components/PromoBanner.tsx (hunk 8 above)
Comment (dahaca) [reply packages/suite/src/views/dashboard/components/PromoBanner.tsx]: done

**Review by sime — APPROVED** [review]
App store buttons react to click/touch event on desktop/mobile ✅

**Pull request merged**